modules/effects/src/main/scala/dev/profunktor/redis4cats/redis.scala

```diff
@@ -1230,6 +1230,9 @@ private[redis4cats] class BaseRedis[F[_]: Concurrent: ContextShift, K, V](
   override val ping: F[String] =
     async.flatMap(c => F.delay(c.ping())).futureLift
 
+  override def select(index: Int): F[Unit] =
+    conn.async.flatMap(c => blocker.delay(c.select(index))).void
```

Comment:
Do we need to use `Blocker` here? The async Lettuce API should not need it

Reply:
Yes it seems that the `blocker` is needed for now. Even though this is the async api the `select` issues a command to redis and waits/blocks for the response.

Here is the relevant lettuce code
```
public String select(int db) {
    AsyncCommand<K, V, String> cmd = selectAsync(db);
    return LettuceFutures.awaitOrCancel(cmd, connection.getTimeout().toNanos(), TimeUnit.NANOSECONDS);
}
```
Under the hood, `awaitOrCancel` delegates to `CompletableFuture::get()`.

This inconvenience should be solved in the upcoming release of lettuce: [relevant commit](https://github.com/lettuce-io/lettuce-core/commit/426fb0cf0326e74f27aa4feacc7f8828ce1a9337)

Reply:
Good to know, thanks!